src/ripple/app/hook/applyHook.h

```diff
@@ -888,12 +1288,396 @@ class HookExecutor
+            ADD_JS_FUNCTION(etxn_nonce, ctx);
+            
+            ADD_JS_FUNCTION(float_set, ctx);
+            ADD_JS_FUNCTION(float_multiply, ctx);
+            ADD_JS_FUNCTION(float_mulratio, ctx);
+            ADD_JS_FUNCTION(float_negate, ctx);
+            ADD_JS_FUNCTION(float_compare, ctx);
+            ADD_JS_FUNCTION(float_sum, ctx);
+            ADD_JS_FUNCTION(float_sto, ctx);
+            ADD_JS_FUNCTION(float_sto_set, ctx);
+            ADD_JS_FUNCTION(float_invert, ctx);
+
+            ADD_JS_FUNCTION(float_divide, ctx);
+            ADD_JS_FUNCTION(float_one, ctx);
+            ADD_JS_FUNCTION(float_mantissa, ctx);
+            ADD_JS_FUNCTION(float_sign, ctx);
+            ADD_JS_FUNCTION(float_int, ctx);
+            ADD_JS_FUNCTION(float_log, ctx);
+            ADD_JS_FUNCTION(float_root, ctx);
+
+            ADD_JS_FUNCTION(otxn_burden, ctx);
+            ADD_JS_FUNCTION(otxn_generation, ctx);
+            ADD_JS_FUNCTION(otxn_field, ctx);
+            ADD_JS_FUNCTION(otxn_id, ctx);
+            ADD_JS_FUNCTION(otxn_type, ctx);
+            ADD_JS_FUNCTION(otxn_slot, ctx);
+            ADD_JS_FUNCTION(otxn_param, ctx);
+
+            
+            ADD_JS_FUNCTION(hook_account, ctx);
+            ADD_JS_FUNCTION(hook_hash, ctx);
+            ADD_JS_FUNCTION(hook_again, ctx);
+            ADD_JS_FUNCTION(fee_base, ctx);
+            ADD_JS_FUNCTION(ledger_seq, ctx);
+            ADD_JS_FUNCTION(ledger_last_hash, ctx);
+            ADD_JS_FUNCTION(ledger_last_time, ctx);
+            ADD_JS_FUNCTION(ledger_nonce, ctx);
+            ADD_JS_FUNCTION(ledger_keylet, ctx);
+            
+            ADD_JS_FUNCTION(hook_param, ctx);
+            ADD_JS_FUNCTION(hook_param_set, ctx);
+            ADD_JS_FUNCTION(hook_skip, ctx);
+            ADD_JS_FUNCTION(hook_pos, ctx);
+
+            ADD_JS_FUNCTION(state, ctx);
+            ADD_JS_FUNCTION(state_foreign, ctx);
+            ADD_JS_FUNCTION(state_set, ctx);
+            ADD_JS_FUNCTION(state_foreign_set, ctx);
+
+            ADD_JS_FUNCTION(slot, ctx);
+            ADD_JS_FUNCTION(slot_clear, ctx);
+            ADD_JS_FUNCTION(slot_count, ctx);
+            ADD_JS_FUNCTION(slot_set, ctx);
+            ADD_JS_FUNCTION(slot_size, ctx);
+            ADD_JS_FUNCTION(slot_subarray, ctx);
+            ADD_JS_FUNCTION(slot_subfield, ctx);
+            ADD_JS_FUNCTION(slot_type, ctx);
+            ADD_JS_FUNCTION(slot_float, ctx);
+
+            ADD_JS_FUNCTION(trace, ctx);
+            /*
+            ADD_JS_FUNCTION(trace_num, ctx);
+            ADD_JS_FUNCTION(trace_float, ctx);
+
+            */
+            ADD_JS_FUNCTION(meta_slot, ctx);
+            ADD_JS_FUNCTION(xpop_slot, ctx);
+        }
+
+        bool
+        sane()
+        {
+            return ctx && rt;
+        }
+
+        ~QuickJSVM()
+        {
+            if (rt)
+                JS_FreeRuntime(rt);
+            if (ctx)
+                JS_FreeContext(ctx);
+            rt = NULL;
+            ctx = NULL;
```

Comment:
In my environment, Object Leaks error occurred when running a local build, but changing it as suggested resolved the error.

Also, in the [quickjs sample code](https://github.com/bellard/quickjs/blob/6e2e68fd0896957f92eb6c242a2e048c1ef3cae0/fuzz/fuzz_eval.c#L46-L47), JS_FreeRuntime is called after JS_FreeContext, as suggested.

```suggestion
            if (ctx)
                JS_FreeContext(ctx);
            if (rt)
                JS_FreeRuntime(rt);
            ctx = NULL;
            rt = NULL;
```